File: app/src/main/java/org/metabrainz/mobile/presentation/view/RobotoLightTextView.java
package org.metabrainz.mobile.presentation.view;

import android.content.Context;
import android.graphics.Typeface;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

import org.metabrainz.mobile.App;

public class RobotoLightTextView extends AppCompatTextView {

    public RobotoLightTextView(Context context) {
        super(context);
    }

    public RobotoLightTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public RobotoLightTextView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public void setTypeface(Typeface tf) {
        if (isInEditMode()) {
            super.setTypeface(tf);
        } else {
            super.setTypeface(App.getRobotoLight());
        }
    }

}

File: app/src/main/java/org/metabrainz/mobile/presentation/view/RobotoLightTextView.kt
package org.metabrainz.mobile.presentation.view

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import org.metabrainz.mobile.App

class RobotoLightTextView : AppCompatTextView {
    constructor(context: Context?) : super(context!!)
    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(context!!, attrs, defStyle)

    override fun setTypeface(tf: Typeface?) {
        if (isInEditMode) {
            super.setTypeface(tf)
        } else {
            super.setTypeface(App.getRobotoLight())
        }
    }
}